tools/load_vcvarsall.sh

```diff
@@ -0,0 +1,64 @@
+# You can source it or add this to your ~/.bashrc for example
+
+# call load_msenv [year] from a shell to load the x64 enviroment for the corresponding VS year.
+# it will cache the variables in $HOME/.msenv${msversion_year}_bash. Delete it if you want to reload.
+function load_msenv() {
+    local msversion_year=2022
+    if [ $# -gt 0 ]; then
+        msversion_year=$1
+    fi
+
+    case $msversion_year in
+        2017)
+            msversion_prod=15
+            ;;
+        2019)
+            msversion_prod=16
+            ;;
+        2022)
+            msversion_prod=17
+            ;;
+        *)
+            >&2 printf "Invalid version year. Supported are 2017, 2019 and 2022."
```

Comment:
do we still support VS 2017? It does not (fully) support C17, right?

Reply:
I think not, yes

Reply:
but the script might be helpful for older versions as well.